File: admin-app/src/main/java/com/example/admin/web/CatalogService.java
package com.example.admin.web;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RestController;

import com.example.admin.dto.ProductDto;

import reactor.core.publisher.Mono;

@RestController
public class CatalogService {

	@Value("${example.products-app.name}")
	private String productsApp;
	@Value("${example.products-app.getProductUrl}")
	private String getProductUrl;

	@Autowired
	private WebClientTemplate client;

	@GetMapping("products/get/{id}")
	private Mono<ProductDto> getProduct(@PathVariable String id) {
		return client.get(productsApp, getProductUrl.replace("{id}", id), ProductDto.class);
	}
}

File: admin-app/src/main/kotlin/com/example/admin/web/CatalogService.kt
package com.example.admin.web

import com.example.admin.dto.ProductDto
import org.springframework.cloud.client.discovery.DiscoveryClient
import org.springframework.http.MediaType
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RestController
import org.springframework.web.reactive.function.client.WebClient
import reactor.core.publisher.Mono
import java.util.*

@RestController
class CatalogService (private val wct: WebClientTemplate) {

    @GetMapping("products/get/{id}")
    fun getProduct(@PathVariable id: String): Mono<ProductDto> {
        return wct.get(
                "products-app",
                "products/get/" + id,
                ProductDto::class.java)
    }
}
